web-server/src/components/Settings/SyncDaysSetting.tsx

```diff
@@ -0,0 +1,146 @@
+import { Line } from '../Text';
+import { FC, useEffect, useState } from 'react';
+import { useAuth } from '@/hooks/useAuth';
+import { useSnackbar } from 'notistack';
+import { useDispatch, useSelector } from '@/store';
+import {
+  getDefaultSyncDaysSettings,
+  updateDefaultSyncDaysSettings
+} from '@/slices/org';
+import { FetchState } from '@/constants/ui-states';
+
+export const SyncDaysSetting: FC = () => {
+  const dispatch = useDispatch();
+  const { orgId } = useAuth();
+  const { enqueueSnackbar } = useSnackbar();
+  const initialSyncDays = useSelector((state) => state.org.defaultSyncDays);
+  const [syncDays, setSyncDays] = useState<number>(initialSyncDays);
+  const isSaving = useSelector(
+    (s) => s.org.requests?.defaultSyncDays === FetchState.REQUEST
+  );
+  const [error, setError] = useState('');
+
+  const getSyncDays = async (): Promise<number> => {
+    const res = await dispatch(getDefaultSyncDaysSettings({ orgId }));
+    return res.payload as number;
+  };
+
+  const updateSyncDays = async (days: number): Promise<number> => {
+    const res = await dispatch(
+      updateDefaultSyncDaysSettings({ orgId, defaultSyncDays: days })
+    );
+    return res.payload as number;
+  };
+
+  const handleDiscard = () => {
+    setSyncDays(initialSyncDays);
+    setError('');
+  };
+
+  const handleSave = async () => {
+    try {
+      await updateSyncDays(syncDays);
+      enqueueSnackbar('Default Sync Days Updated Successfully', {
+        variant: 'success',
+        autoHideDuration: 3000
+      });
+    } catch (e) {
+      handleDiscard();
+      return enqueueSnackbar('Default Sync Days Update Failed', {
+        variant: 'error',
+        autoHideDuration: 2000
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (!orgId) return;
+    getSyncDays().then((d) => setSyncDays(d));
+  }, []);
+
+  return (
+    <FlexBox col gap2>
+      <FlexBox big bold gap1 component={Line} white sx={{ fontSize: '20px' }}>
+        Organization Settings
+      </FlexBox>
+
+      <Divider />
+      {initialSyncDays && (
+        <>
+          <FlexBox gap1 alignCenter justifyBetween>
+            <FlexBox col flexGrow={1}>
+              <Line big bold>
+                Sync Days
+              </Line>
+              <Box>
+                The setting to control how many days of past data should be
+                included.
+              </Box>
+            </FlexBox>
+
+            <FlexBox alignCenter>
+              <TextField
+                value={syncDays}
+                onChange={(e) => {
+                  const days = Number(e.target.value);
+                  if (isNaN(days)) setError('Sync days is Not a number');
+                  else if (days < initialSyncDays)
+                    setError('Less than Current Sync Days not allowed');
+                  else if (days > 366)
+                    setError('Sync days cannot be more than 366');
+                  else setError('');
+                  setSyncDays(days);
```

Comment:
@Kamlesh72 I do not understand the decision, to not allow the user to change default sync days less than current setting. If someone saves 366, I think they should be allowed to update the default sync days to a lower value if they wanted. The minimum value for sync days can be 31 days. Also pls add 366 and 31 days in const variables.
![image](https://github.com/user-attachments/assets/6775fc83-6469-42c0-949d-f7d37aace820)

Reply:
also, I see we are not calling sync here, so are we gonna leave it to the cron to sync data on the next sync cycle ?

Reply:
We can final it after refactor of settings page.

Reply:
@samad-yar-khan is this fine? I will make the commit.

Reply:
@Kamlesh72 Yup please do. Also, lets call sync after setting updation.

Reply:
It would be super convenient to have data synced as soon as settings are saved, but we will have to see how to handle this incase settings are changed again and again.

Reply:
@samad-yar-khan as of now, do we need to show some modal to sync?

Reply:
@Kamlesh72 You can ignore my comments about the sync date range, had a talk @jayantbh . He agrees with your logic. The only change I would recommend in this PR is triggering sync after setting change.

Reply:
@samad-yar-khan We are triggering sync without asking user, right?

Reply:
yes, for now